src/domain/tagset/tagset.resolver.ts

```diff
@@ -17,35 +15,35 @@ export class TagsetResolver {
     RestrictedGroupNames.EcoverseAdmins
   )
   @UseGuards(GqlAuthGuard)
-  @Mutation(() => Tagset, {
+  @Mutation(() => Boolean, {
     description: 'Replace the set of tags in a tagset with the provided tags',
   })
   async replaceTagsOnTagset(
     @Args('tagsetID') tagsetID: number,
-    @Args('tags') newTags: TagsInput
-  ): Promise<ITagset> {
-    if (!newTags.tags)
+    @Args({ name: 'tags', type: () => [String] }) newTags: string[]
+  ): Promise<boolean> {
+    if (!newTags)
       throw new Error(`Unable to replace tags on tagset(${tagsetID}`);
 
-    const tagset = await this.tagsetService.replaceTags(tagsetID, newTags.tags);
+    await this.tagsetService.replaceTags(tagsetID, newTags);
 
-    return tagset;
+    return true;
   }
 
   @Roles(
     RestrictedGroupNames.CommunityAdmins,
     RestrictedGroupNames.EcoverseAdmins
   )
   @UseGuards(GqlAuthGuard)
-  @Mutation(() => Tagset, {
+  @Mutation(() => Boolean, {
```

Comment:
The result of a mutation must return object that contains a primary key. It is crucial to have the Identifier in the result, because the ApolloClient cache is using it to updates its internal cache.